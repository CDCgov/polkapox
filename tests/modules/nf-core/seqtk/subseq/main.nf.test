nextflow_process {

    name "Test Process SEQTK_SUBSEQ"
    script "modules/nf-core/seqtk/subseq/main.nf"
    process "SEQTK_SUBSEQ"

    test("SEQTK Test") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                        [meta: 'test_sub'], 
                        [file('${projectDir}/tests/nf-test-data/test_sub_1.fq.gz'), file('${projectDir}/tests/nf-test-data/test_sub_2.fq.gz')]
                        ]
                input[1] = [ [meta: 'test_sub'], '${projectDir}/tests/nf-test-data/test_sub.kraken2.classifiedreads.txt' ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
