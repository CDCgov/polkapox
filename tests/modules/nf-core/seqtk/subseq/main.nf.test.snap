{
    "SEQTK Test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "opxv_reads": [
                    
                ],
                "reads": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:21:35.521133148"
    }
}